File: Atividade1/Ruby/main.rb
# implementação da função strcmp
def strcmp(str1, str2)
    if !str1.is_a?(String) || !str2.is_a?(String)
        raise "Ambas as entradas devem ser strings"
    end

    (str1 > str2) ? 1 : ((str1 < str2) ? -1 : 0)
end

# implementação da função strcat
def strcat(str1, str2)
    if !str1.is_a?(String) || !str2.is_a?(String)
        raise "Ambas as entradas devem ser strings"
    end
    result = str1 + str2
end


string1 = 1
string2 = "ABC"

result = strcmp(string1, string2) # chamada da função strcmp

if result > 0
    puts "#{string1} é maior que #{string2} na tabela ascii"
elsif result < 0
    puts "#{string1} é menor que #{string2} na tabela ascii"
else
    puts "#{string1} é igual a #{string2} na tabela ascii"
end

stringAux = strcat(string1, string2) # chamada da função strcat
puts "#{stringAux} é a concatenação de #{string1} e #{string2}"

File: Atividade1/JavaScript/main.js
// implementação da função strcmp
function strcmp(str1, str2) {
    if (typeof str1 !== 'string' || typeof str2 !== 'string') 
        throw Error("Ambas as entradas devem ser strings")

    return (str1 > str2) - (str1 < str2)
}

// implementação da função strcat
function strcat(str1, str2){
    if (typeof str1 !== 'string' || typeof str2 !== 'string') 
        throw Error("Ambas as entradas devem ser strings")
    
    return str1 + str2;
}

var string1 = "abc"
var string2 = "ABC"

var result = strcmp(string1, string2) // chamada da função strcmp

if (result > 0)
    console.log("%s é maior que %s na tabela ascii", string1, string2)
else if (result < 0)
    console.log("%s é menor que %s na tabela ascii", string1, string2)
else
    console.log("%s é igual a %s na tabela ascii", string1, string2)


stringAux = strcat(string1, string2) // chamada da função strcat
console.log("%s é a concatenação de %s e %s", stringAux, string1, string2)